### PR TITLE
Use argument and return type line numbers for error messages

### LINKED ISSUE #3177: Invalid definition of Tuple in return reports on a confusing line number
Here's an example:
```
class TestTuple:
    @staticmethod
    def test(a: str) -> (bool, str):
        return len(a) < 2, a[:2]
```
And here's the output (I'd expect it to say Line #3):
```
test_invalid_tuple.py:2: error: Invalid tuple literal type
```

### PR DESCRIPTION
**Pull request by patrickw276**
Fixes #3177

### CODE REVIEW

**Comment by gvanrossum** [comment]
I'll review.

File: test-data/unit/check-functions.test [hunk 1]
```diff
@@ -2044,3 +2044,16 @@ bad = make_list()  # E: Need type annotation for variable
 def foo(__b: int, x: int, y: int) -> int: pass
 foo(x=2, y=2)  # E: Missing positional argument
 foo(y=2)  # E: Missing positional arguments
+
+[case testReturnTypeLineNumberWithDecorator]
+def dec(f): pass
+
+@dec
+def test(a: str) -> (str,): # E: Invalid tuple literal type
```
Comment (gvanrossum) [inline test-data/unit/check-functions.test]: Can you add a test where the `) -> ...` part is on a separate line?

File: test-data/unit/check-functions.test [hunk 2]
```diff
@@ -2044,3 +2044,16 @@ bad = make_list()  # E: Need type annotation for variable
...
+    return None
+
+[case testArgumentTypeLineNumberWithDecorator]
+def dec(f): pass
+
+@dec
+def some_method(self: badtype): pass # E: Name 'badtype' is not defined
```
Comment (gvanrossum) [inline test-data/unit/check-functions.test]: imilar with multiple arguments across multiple lines.

Comment (patrickw276) [inline test-data/unit/check-functions.test]: Updated PR with additional tests.

**Pull request merged**